presto-main/src/main/java/com/facebook/presto/security/FileBasedSystemAccessControl.java

```diff
@@ -80,10 +84,14 @@ public SystemAccessControl create(Map<String, String> config)
                 }
                 path.toFile().canRead();
 
+                FileBasedSystemAccessControlRules rules = jsonCodec(FileBasedSystemAccessControlRules.class)
+                        .fromJson(Files.readAllBytes(path));
+
                 ImmutableList.Builder<CatalogAccessControlRule> catalogRulesBuilder = ImmutableList.builder();
-                catalogRulesBuilder.addAll(jsonCodec(FileBasedSystemAccessControlRules.class)
-                        .fromJson(Files.readAllBytes(path))
-                        .getCatalogRules());
+                catalogRulesBuilder.addAll(rules.getCatalogRules());
+
+                ImmutableList.Builder<Pattern> usernameExtractionRulesBuilder = ImmutableList.builder();
+                usernameExtractionRulesBuilder.addAll(rules.getUsernameExtractionRules());
```

Comment:
will `getUsernameExtractionRules()` return empty list if user does not specify it in `json` file. 
Or will this throw `NPE`.
Maybe add test for parsing rules file.